e2etest/bgptests/bgp.go

```diff
@@ -106,113 +106,86 @@ var _ = ginkgo.Describe("BGP", func() {
 		cs = f.ClientSet
 	})
 
-	table.DescribeTable("A service of protocol load balancer should work with", func(pairingIPFamily ipfamily.Family, setProtocoltest string, poolAddresses []string, tweak testservice.Tweak) {
-		var allNodes *corev1.NodeList
-		resources := metallbconfig.ClusterResources{
-			Pools: []metallbv1beta1.IPAddressPool{
-				{
-					ObjectMeta: metav1.ObjectMeta{
-						Name: "bgp-test",
-					},
-					Spec: metallbv1beta1.IPAddressPoolSpec{
-						Addresses: poolAddresses,
-					},
-				},
-			},
-		}
-
-		err := ConfigUpdater.Update(resources)
-		framework.ExpectNoError(err)
+	table.DescribeTable("A service of protocol load balancer should work with ETP=cluster", func(pairingIPFamily ipfamily.Family, poolAddresses []string, tweak testservice.Tweak) {
 
-		svc, jig := testservice.CreateWithBackend(cs, f.Namespace.Name, "external-local-lb", tweak)
+		_, svc := setupBGPService(f, pairingIPFamily, poolAddresses, func(svc *corev1.Service) {
+			testservice.TrafficPolicyCluster(svc)
+			tweak(svc)
+		})
 		defer testservice.Delete(cs, svc)
```

Comment:
the defer to delete the service is set twice, once here and once in the setupBGPService func

Reply:
i see it's removed in the third commit but in the first commit it is still done twice